### PR TITLE
feat(checkout): CHECKOUT-7255 Remove checkout step numbers

### PR DESCRIPTION
**Pull request by bc-peng**
## What?
Removing all checkout step numbers.
- Remove the step counter of each unfinished checkout step.
- Remove the circle, check mark of each finished checkout step.
- Change the step header’s left padding to zero.
- Putting all changes behind an experiment so that we can roll out changes incrementally.

## Why?
Once we introduce wallet buttons on the top of the checkout page, the step numbers are no longer accurate progress indicators.

## Testing / Proof

customer

shipping

payment

https://user-images.githubusercontent.com/88361607/216851440-9d7db6c9-6f80-4427-8cd9-02e18ff33442.mov

@bigcommerce/checkout

### CODE REVIEW

File: packages/core/src/app/checkout/Checkout.tsx [hunk 1]
```diff
@@ -286,7 +292,7 @@ class Checkout extends Component<
         }
 
         return (
-            <div className={classNames({ 'is-embedded': isEmbedded() })}>
+            <div className={classNames({ 'is-embedded': isEmbedded(), 'remove_checkout_step_numbers': isHidingStepNumbers })}>
```
Comment (bc-peng) [inline packages/core/src/app/checkout/Checkout.tsx:295]: Initially, I changed the html structure as well, but then I realised I could control every UI element with pure CSS if there was a specific class name at the top level.

Comment (animesh1987) [inline packages/core/src/app/checkout/Checkout.tsx:295]:
```suggestion
            <div className={classNames({ 'is-embedded': isEmbedded(), 'remove-checkout-step-numbers': isHidingStepNumbers })}>
```

Comment (animesh1987) [reply packages/core/src/app/checkout/Checkout.tsx:295]: Also should we use `remove-checkout-step-numbers` ?

Comment (bc-peng) [reply packages/core/src/app/checkout/Checkout.tsx:295]: I believe we will remove this class name this or next month, so it is only there to support the experiment's roll-out.
Just want to confirm, are you suggesting changing a name or not to use this approach?

Comment (animesh1987) [reply packages/core/src/app/checkout/Checkout.tsx:295]: Just changing the name.

Comment (bc-peng) [reply packages/core/src/app/checkout/Checkout.tsx:295]: What name(s) do you have in mind?

Comment (bc-peng) [reply packages/core/src/app/checkout/Checkout.tsx:295]: Since it is most likely for us to have a toggle on this, maybe `is-hiding-step-numbers`?

**Comment by bc-peng** [comment]
Hi @animesh1987, I have updated the code and tested it on my store. Can I have another review?

File: packages/core/src/scss/components/checkout/checkoutSteps/_checkoutSteps.scss [hunk 2]
```diff
@@ -138,6 +138,21 @@
...
+        div.stepHeader-counter.icon {
+            display: none;
+        }
+
+        .checkout-form,
+        div.stepHeader-body,
+        div.checklist-skeleton,
+        h2.stepHeader-title {
+            margin-left: 0;
+        }
```
Comment (animesh1987) [inline packages/core/src/scss/components/checkout/checkoutSteps/_checkoutSteps.scss:152]: Looks good, but just wondering do we need to add `div, li, h2` as a part of css identifiers?

Comment (bc-peng) [reply packages/core/src/scss/components/checkout/checkoutSteps/_checkoutSteps.scss:152]: By doing so, I feel confident they're at the top of the checkout CSS hierarchy. 😂

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**